pywhat/identifier.py

```diff
@@ -1,36 +1,64 @@
-
-class Identifier:
-    def __init__(self):
-        self.regex_id = RegexIdentifier()
-        self.file_sig = FileSignatures()
-        self.name_that_hash = Nth()
-
-    def identify(self, text: str, api=False) -> dict:
-        identify_obj = {}
-
-        magic_numbers = None
-        if not api and self.file_exists(text):
-            magic_numbers = self.file_sig.open_binary_scan_magic_nums(text)
-            text = self.file_sig.open_file_loc(text)
-            identify_obj["File Signatures"] = magic_numbers
-        else:
-            text = [text]
-
-        if not magic_numbers:
-            # If file doesn't exist, check to see if the inputted text is
-            # a file in hex format
-            identify_obj["File Signatures"] = self.file_sig.check_magic_nums(text)
-        identify_obj["Regexes"] = self.regex_id.check(text)
-        # get_hashes takes a list of hashes, we split to give it a list
-        # identify_obj["Hashes"] = self.name_that_hash.get_hashes(text.split())
-
-        return identify_obj
-
-    def file_exists(self, text):
-        return os.path.isfile(text)
+import os.path
+import glob
+
+from pywhat.magic_numbers import FileSignatures
+from pywhat.nameThatHash import Nth
+from pywhat.regex_identifier import RegexIdentifier
+
+
+class Identifier:
+    def __init__(self):
+        self.regex_id = RegexIdentifier()
+        self.file_sig = FileSignatures()
+        self.name_that_hash = Nth()
+
+    def identify(self, input: str, api=False) -> dict:
+        identify_obj = {}
+        identify_obj["File Signatures"] = {}
+        identify_obj["Regexes"] = {}
+
+        if os.path.isdir(input):
+            # if input is a directory, recursively search for all of the files
+            for myfile in glob.iglob(input + "**/**", recursive=True):
+                if os.path.isfile(myfile):
+                    magic_numbers = self.file_sig.open_binary_scan_magic_nums(myfile)
+                    text = self.file_sig.open_file_loc(myfile)
+
+                    if not magic_numbers:
+                        magic_numbers = self.file_sig.check_magic_nums(myfile)
+
+                    short_name = os.path.basename(myfile)
+                    identify_obj["File Signatures"][short_name] = magic_numbers
+                    identify_obj["Regexes"][short_name] = self.regex_id.check(text)
+
+
+        elif os.path.isfile(input):
+            short_name = os.path.basename(input)
+            magic_numbers = self.file_sig.open_binary_scan_magic_nums(input)
+            text = self.file_sig.open_file_loc(input)
+
+            # if file doesn't exist, check to see if the inputted text is
+            # a file in hex format
+            if not magic_numbers:
+                magic_numbers = self.file_sig.check_magic_nums(text)
+
+            short_name = os.path.basename(input)
+            identify_obj["File Signatures"][short_name] = magic_numbers
+            identify_obj["Regexes"][short_name] = self.regex_id.check(text)
```

Comment:
You wrote essentially the same code twice. Maybe you can shorten it? 😅

Reply:
Well this was in a function before but I changed it after this two comments from Bee: [1](https://github.com/bee-san/pyWhat/pull/58#discussion_r643007439) and [2](https://github.com/bee-san/pyWhat/pull/58#discussion_r643008060)

Reply:
Well, you can store all filenames in a list and iterate over it. This way you will not be repeating yourself twice.

Reply:
That's a good idea yeah. Will change it later. Thanks! :D

Reply:
I have changed this now and it looks much better 😄

Reply:
Beautiful!